### PR TITLE
Limiting team/repository access for users inactive for 60 months

### PR DESCRIPTION
**Pull request by galargh**
### Summary

This PR cleans up user access by removing users who have been inactive for over 60 months (5 years) from teams and repositories.

A user is deemed inactive if they haven't performed any of the following actions in the past 60 months in a repository in question or in any of the repositories the team in question grants access to:
- created an issue or a pull request
- commented on an issue or a pull request
- reviewed a pull request
- commented on a commit
- pushed (in particular, force pushed) to a branch
- created a branch
- deleted a branch
- merged a pull request
- added a pull request to a merge queue

Any user who, after the introduction of the above changes, isn't a direct collaborator in any of the repositories and isn't a member of any teams is assigned to the `Alumni` team.

If a user's access to a repository or team should be restored, the appropriate line change should be reverted, and a comment starting with `KEEP: ` (followed by a reason) should be added directly above that line.

This pertains to the "'archive' inactive users and teams" in https://github.com/ipfs/ipfs/issues/511.

### Who is this targeting?

The current PR is what results from a [script to identify inactive users in an org](https://github.com/ipfs/github-mgmt/blob/master/scripts/src/actions/remove-inactive-members.ts).

### Why is this being done?
See "Why do we care about periodically cleaning up permissions across the orgs?" in https://github.com/ipfs/ipfs/issues/511

### Is this set in stone?
No. This PR was created and being left open for some days to give awareness and incorporate feedback. We're not taking a "ask for permission" approach, as that would require way too much wrangling. Instead, we're giving visibility to what's proposed and inviting folks to comment and influence. A saving grace here is that none of this is a "one-way door". If something got messed up or missed, a follow-up PR can be done to correct it.

### Is anyone being removed from the organization?
No. All existing members of the org are staying members. In the most reduced/scoped-down case, someone will still be part of an "Alumni" team in the org to signal their past involvement. Thank you for your past contributions, and we certainly welcome you to play a more active role in the future.

### Timeline
2024-02-26: public PR
2024-03-04: notify affected parties with @mention: <insert LINK_TO_COMMENT_IN_THIS_PR>
2024-03-08: merge this change after incorporating feedback
2024-03-08: clean up empty teams and past IPFS Camp teams

### CODE REVIEW

File: github/ipfs.yml [hunk 1]
```diff
@@ -6501,10 +6407,6 @@ teams:
         # Why @mishmosh?
         # 1. Director of the being-stood-up-in-2024 IPFS Foundation
         - mishmosh
-        # Why @stebalien?
-        # 1. Not involved in the IPFS day-to-day currently, but has a lot of historical knowledge.  Provides an informed outside perspective.
-        # 2. Familiar with github-mgmt responsibilities in other orgs.
-        - stebalien
```
Comment (lidel) [inline github/ipfs.yml:6507]: `# KEEP: ` for the reasons above?

File: github/ipfs.yml [hunk 2]
```diff
@@ -6530,24 +6432,13 @@ teams:
...
     description: Who may cut a release of go-ipfs
-    members:
-      member:
-        - aschmahmann
-        - guseggert
-        - lidel
-        - olizilla
-        - petar
-        - Stebalien
     privacy: closed
```
Comment (lidel) [inline github/ipfs.yml:6435]: I think this entire group can be removed, releases were done by people who were not members.

File: github/ipfs.yml [hunk 3]
```diff
@@ -6651,63 +6520,49 @@ teams:
...
         - terichadbourne
         - vasa-develop
         - vojtechsimetka
         - waynewyang
-        - whyrusleeping
         - Wondertan
         - yrliou
-        - zeckli
     privacy: closed
   IPFS Camp 2019 Org Team:
```
Comment (lidel) [inline github/ipfs.yml:6558]: Remove entire team, will save us noise over time.

File: github/ipfs.yml [hunk 4]
```diff
@@ -6741,137 +6595,9 @@ teams:
         - stefanhans
     privacy: closed
   IPFS Camp 2019:
```
Comment (lidel) [inline github/ipfs.yml:6597]: Remove entire team, will save us noise over time.

File: github/ipfs.yml [hunk 5]
```diff
@@ -6741,137 +6595,9 @@ teams:
...
-        - vasa-develop
-        - vojtechsimetka
-        - waynewyang
-        - whyrusleeping
-        - Wondertan
-        - yrliou
-        - yunigraham
-        - zeckli
     privacy: closed
   IPFS HTTP Client Maintainers:
```
Comment (lidel) [inline github/ipfs.yml:6599]: @galargh should we remove teams which end up with no members?
There is a bunch of them below (`Repos - Python`, `Repos - Rust`, `Repos - Scala` etc).
It does not have to be as part of this cleanup, ideally would be separate PR, and also follow `# KEEP: ` convention.

**Review by lidel — COMMENTED** [review]
Thank you, I eyeballed and nothing controversial stands out.  We have a bunch of repos where the original creator still had admin but was not contributing for years. I think it makes sense to remove admin in these cases  as a security precaution  (can be restored if they need it back or want to transfer repo – example: https://github.com/ipfs/stargate/issues/7). 

Some comments inline.

File: github/ipfs.yml (hunk 2 above)
Comment (galargh) [reply github/ipfs.yml:6435]: Let's remove empty teams after the user related changes are complete.

File: github/ipfs.yml (hunk 3 above)
Comment (galargh) [reply github/ipfs.yml:6558]: Noted, I'll remove it in a separate PR.

File: github/ipfs.yml (hunk 5 above)
Comment (galargh) [reply github/ipfs.yml:6599]: Yes, let's remove the empty teams but I'd prefer doing it after this round of user changes is complete if you don't mind.

**Comment by galargh** [comment]
Hi! We're ready to open up the PR for general review 🥳

I'd like to ask you to review the changes affecting you and flag any that should be reverted, are wrong, or need more explanation. You can find the detailed explanation of this PR, the reasoning for introducing the changes and the process itself in the description - https://github.com/ipfs/github-mgmt/pull/193#issue-2153562441

Thank you, and let me know if you have any questions 💁

Tagging all the people whose access changes (https://github.com/ipfs/github-mgmt/pull/193#issuecomment-1963563098) as a result of this PR (no one is being removed from the org):
- [ ] @1015bit
- [ ] @2color
- [ ] @AdamStone
- [ ] @BrendanBenshoof
- [ ] @DrCatman
- [ ] @JGAntunes
- [ ] @JonKrone
- [ ] @Kubuxu
- [ ] @MAkcanca
- [ ] @NeoTeo
- [ ] @Robmat05
- [ ] @ShishKabab
- [ ] @SidHarder
- [ ] @abn
- [ ] @adlrocha
- [ ] @akru
- [ ] @akrych
- [ ] @alekswn
- [ ] @angiemaguire
- [ ] @apronkin
- [ ] @arcalinea
- [ ] @arsstone
- [ ] @asutula
- [ ] @balupton
- [ ] @cleichner
- [ ] @cmk
- [ ] @cwaring
- [ ] @cyborgshead
- [ ] @damedoteth
- [ ] @dankelleher
- [ ] @darkdh
- [ ] @davidar
- [ ] @daviddias
- [ ] @dborzov
- [ ] @dchoi27
- [ ] @deltazxm
- [ ] @dignifiedquire
- [ ] @dongtianyi
- [ ] @dryajov
- [ ] @elijaharita
- [ ] @emerick
- [ ] @eminence
- [ ] @ericronne
- [ ] @eshon
- [ ] @fbaiodias
- [ ] @flyingzumwalt
- [ ] @gavinmcdermott
- [ ] @gfanton
- [ ] @gmasgras
- [ ] @guillemcordoba

1/3

**Comment by galargh** [comment]
Continuation of https://github.com/ipfs/github-mgmt/pull/193#issuecomment-1977089729

- [ ] @guseggert
- [ ] @haadcode
- [ ] @hannahhoward
- [ ] @harlantwood
- [ ] @hosh
- [ ] @hugomrdias
- [ ] @ipfs-gui-bot
- [ ] @ipfsbot
- [ ] @jacobheun
- [ ] @jamiejn
- [ ] @jbenet
- [ ] @jsoares
- [ ] @juliaxbow
- [ ] @kinu
- [ ] @krl
- [ ] @kumavis
- [ ] @lidel
- [ ] @litzenberger
- [ ] @locotorp
- [ ] @ma82
- [ ] @magik6k
- [ ] @marcooliveira
- [ ] @marshall
- [ ] @mcollina
- [ ] @mildred
- [ ] @mishmosh
- [ ] @moeghashim
- [ ] @momack2
- [ ] @neogeweb3
- [ ] @nicola
- [ ] @nunofmn
- [ ] @parkan
- [ ] @petar
- [ ] @pooja
- [ ] @ratoshniuk
- [ ] @raulk
- [ ] @sanderpick
- [ ] @sericaia
- [ ] @smihaylov
- [ ] @stongo
- [ ] @tbenbow
- [ ] @teh-f00l
- [ ] @tikagan
- [ ] @travisperson
- [ ] @vmx
- [ ] @vyzo
- [ ] @whyrusleeping
- [ ] @yangwao
- [ ] @yiannisbot
- [ ] @yunigraham

2/3

**Comment by galargh** [comment]
Continuation of https://github.com/ipfs/github-mgmt/pull/193#issuecomment-1977089729

- [ ] @yusefnapora
- [ ] @zeckli

3/3

**Comment by vyzo** [comment]
Is this for ipfs only?
 I am still very much involved in libp2p.

**Comment by galargh** [comment]
> Is this for ipfs only?
> I am still very much involved in libp2p.

For you, as per https://github.com/ipfs/github-mgmt/pull/193#issuecomment-1963563098, this will result in the following change:

```
User vyzo:
  - will lose admin permission to go-bitswap-priv
```

**Comment by vyzo** [comment]
ah, ok. Thanks.

**Comment by jsoares** [comment]
Understood, no objections from my side. Thanks for the heads-up!

**Review by 2color — APPROVED** [review]
No objections regarding changes affecting me

**Review by jsoares — APPROVED** [review]

**Review by jsoares — APPROVED** [review]

**Review by vmx — APPROVED** [review]

**Review by ma82 — APPROVED** [review]

**Review by hugomrdias — APPROVED** [review]

File: github/ipfs.yml [hunk 6]
```diff
@@ -340,9 +340,6 @@ repositories:
     advanced_security: false
```
Comment (cwaring) [inline github/ipfs.yml:340]: @galargh feel free to drop the entire `2022.ipfs.camp` key, it was an empty repo that I've since removed. ty

Comment (galargh) [reply github/ipfs.yml:340]: Noted, thanks. I'll clean up teams I was told are no longer relevant after this one's merged.

**Review by JGAntunes — APPROVED** [review]
